### ISSUE TITLE
Keyword-only arguments confuse annotation parsing

### ISSUE THREAD

**Issue by alexjurkiewicz** [PyCQA/pylint#1168]
### Steps to reproduce
```
$ cat test.py
from typing import Optional, Callable, Iterable

def func1(arg: Optional[Callable]=None):
    return arg

def func2(*,
          arg: Optional[Iterable]=None):
    return arg
```

### Current behavior

```
$ pylint -r no test.py
No config file found, using default configuration
************* Module test
C:  1, 0: Missing module docstring (missing-docstring)
C:  3, 0: Missing function docstring (missing-docstring)
C:  6, 0: Missing function docstring (missing-docstring)
W:  1, 0: Unused Iterable imported from typing (unused-import)
```

### Expected behavior

Iterable should be detected as being used.

### pylint --version output

```
$ pylint --version
No config file found, using default configuration
pylint 1.6.4,
astroid 1.4.8
Python 3.5.1 (default, Apr 18 2016, 11:46:32)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.29)]
```

**Comment by PCManticore** [comment]
Thank you, I can reproduce the issue.

**Comment by petrpulc** [comment]
Cannot reproduce anymore in master. (Fixed?)

**Comment by PCManticore** [comment]
This is fixed now with astroid's master, will be part of 1.7.0.

**Comment by alexjurkiewicz** [comment]
👍👍👍👍 Nice! Thanks!
